### PR TITLE
Fixed IPv4 to IPv6 mapping in CIPAddress::pton(..)

### LINKED ISSUE #1663: Fix a wrong IP4-mapped-IP6 assignation in CIPAddress::pton
If I call _srt_connect_ to the IP4-mapped-IP6 address **127.0.0.1** and on success I call _srt_getsockname_
It returns a **0.0.0.0** address since 1.4.2 SRT version (worked in 1.4.1).

Is just a careless error on an index assignment here =>
https://github.com/Haivision/srt/blob/b0a9d4df45f828a9e9f57319b63312ae15b96384/srtcore/common.cpp#L457

Fix:
`paddr32[3] = ip[3]`

### PR DESCRIPTION
**Pull request by maxsharabayko**
While converting IPv4 mapped to IPv6 there was an error in the IPv4 index.

Fixes #1663 

Added unit test for the case.

### TODO

- [x] Unit test to get 100% coverage of `CIPAddress::pton(..)`

### CODE REVIEW

**Comment by maxsharabayko** [comment]
### TestIPv6.v4_calls_v6_mapped
 
#### Listener
 
```shell
# On HS Conclusion Request:
# CUDT::acceptAndRespond(..)
 
agent_family: IPv6 (m_SelfAddr.family)
peer: "::ffff:127.0.0.1:64888"
ip[4]: 16777343, 0, 0, 0

# After CIPAddress::pton():
m_SelfAddr: "::ffff:127.0.0.1:4200"

#CIPAddress::ntop(peer, (w_hs.m_piPeerIP));
w_hs.m_piPeerIP[4]: 0, 0, 0xFF, 16777343
# Goes to HS Conclusion Response
```

#### Caller

```shell
# On HS Conclusion Response:
# CUDT::postConnect(..)

m_SelfAddr.family() (agent_family): IPv4 (AF_INET)
ip[4]: 0, 0, 0xFF, 16777343
peer: "127.0.0.1:4200"

Resulting m_SelfAddr: "0.0.0.0:52872"
```

**Comment by maxsharabayko** [comment]
### TestIPv6.v6_calls_v4

#### Caller Sends HS Induction Request

```shell
Peer IP Address[4]: 0, 0, 0xFF, 16777343
```

#### Listener Sends HS Induction Response:

```shell
Peer IP Address[4]: 0, 0, 0xFF, 16777343
```

#### Caller Sends HS Conclusion Request

```shell
Peer IP Address[4]: 0, 0, 0xFF, 16777343
```

#### Listener Sends HS Conclusion Response ❗ 

```shell
Peer IP Address[4]: 16777343, 0, 0xFF, 16777343
```

File: srtcore/common.cpp [hunk 1]
```diff
@@ -380,83 +388,139 @@ void CIPAddress::ntop(const sockaddr_any& addr, uint32_t ip[4])
...
-      const uint16_t* mbegin = ipv4on6_model;
-      const uint16_t* mend = ipv4on6_model + 6;
+        uint16_t* peeraddr16 = (uint16_t*)peer.sin6.sin6_addr.s6_addr;
+        // Compare only first 6 words. Remaining 2 words
+        // comprise the IPv4 address, if these first 6 match.
+        const uint16_t* mbegin = ipv4on6_model;
+        const uint16_t* mend = ipv4on6_model + 6;
 
-      bool is_mapped_ipv4 = (std::mismatch(mbegin, mend, peeraddr16).first == mend);
+        bool is_mapped_ipv4 = (std::mismatch(mbegin, mend, peeraddr16).first == mend);
```
Comment (maxsharabayko) [inline srtcore/common.cpp:414]:
```c++
const bool is_mapped_ipv4 = std::equal(mbegin, mend, peeraddr16);
```

File: srtcore/common.cpp [hunk 2]
```diff
@@ -380,83 +388,139 @@ void CIPAddress::ntop(const sockaddr_any& addr, uint32_t ip[4])
...
+        // 16-bit:
+        // [0000: 0000: 0000: 0000: 0000: FFFF: 192.168:128.2]
+        // 8-bit
+        // [00/00/00/00/00/00/00/00/00/00/FF/FF/192/168/128/2]
+        // 32-bit
+        // [00000000 && 00000000 && 0000FFFF && 192.168.128.2]
+
+        // Spreading every 16-bit word separately to avoid endian dilemmas
+        paddr16[2 * 2 + 0] = 0;
+        paddr16[2 * 2 + 1] = 0xFFFF;
```
Comment (maxsharabayko) [inline srtcore/common.cpp:473]: `paddr16[2 * 2 + 0] = 0;` ? There was memset to zero right before.

File: .github/workflows/cxx11-win.yaml [hunk 3]
```diff
@@ -21,4 +21,4 @@ jobs:
     - name: build
       run: cd _build && cmake --build ./ --config Release
     - name: test
-      run: cd _build && ctest --extra-verbose -C Release
+      run: cd _build && ctest --extra-verbose -C Release --gtest_gilter="-TestIPv6.v6_calls_v4"
```
Comment (ethouris) [inline .github/workflows/cxx11-win.yaml:24]: gilter? Should that be `filter`?

File: srtcore/common.cpp (hunk 2 above)
Comment (ethouris) [reply srtcore/common.cpp:475]: Might be that it wasn't memset for the full address before, now that it is, indeed this isn't necessary.

**Review by ethouris — COMMENTED** [review]
Minor fixes

**Comment by maxsharabayko** [comment]
Due to CI failures, IPv6 connection tests were extracted to PR #1670 for further fixes, while the bug fix can go within this PR.

**Review by ethouris — APPROVED** [review]

**Pull request merged**